connectors/confluence/connector/src/main/java/org/apache/manifoldcf/crawler/connectors/confluence/client/ConfluenceClient.java

```diff
@@ -0,0 +1,717 @@
+	 * @param url The url identifying the REST resource to get the documents
+	 * @param builder The builder used to build the resources contained in the response
+	 * @return a {@code ConfluenceResponse} containing the page results
+	 * @throws Exception
+	 */
+	private ConfluenceResponse<? extends ConfluenceResource> getConfluenceResources(String url, ConfluenceResourceBuilder<? extends ConfluenceResource> builder) throws Exception {
+		logger.debug("[Processing] Hitting url for get confluence resources: {}", sanitizeUrl(url));
+
+		try {
+			HttpGet httpGet = createGetRequest(url);
+			HttpResponse response = executeRequest(httpGet);
+			ConfluenceResponse<? extends ConfluenceResource> confluenceResponse = responseFromHttpEntity(response
+					.getEntity(), builder);
+			EntityUtils.consume(response.getEntity());
+			return confluenceResponse;
+		} catch (IOException e) {
+			logger.error("[Processing] Failed to get page(s)", e);
+			throw new Exception("Confluence appears to be down", e);
+		}
+	}
+
+	/**
+	 * <p>Creates a ConfluenceResponse from the entity returned in the HttpResponse</p>
+	 * @param entity the {@code HttpEntity} to extract the response from
+	 * @return a {@code ConfluenceResponse} with the requested information
+	 * @throws Exception
+	 */
+	private <T extends ConfluenceResource> ConfluenceResponse<T> responseFromHttpEntity(HttpEntity entity, ConfluenceResourceBuilder<T> builder)
+			throws Exception {
+		String stringEntity = EntityUtils.toString(entity);
+
+		JSONObject responseObject;
+		try {
+			responseObject = new JSONObject(stringEntity);
+			ConfluenceResponse<T> response = ConfluenceResponse
+					.fromJson(responseObject, builder);
+			if (response.getResults().size() == 0) {
+				logger.debug("[Processing] No {} found in the Confluence response", builder.getType().getSimpleName());
+			}
+
+			return response;
+		} catch (JSONException e) {
+			logger.debug("Error parsing JSON response");
+			throw new Exception();
+		}
+
+	}
+	
+	/**
+	 * <p>Get the attachments of the given page</p>
+	 * @param pageId the page id
+	 * @return a {@code ConfluenceResponse} instance containing the attachment results and some pagination values</p>
+	 * @throws Exception
+	 */
+	public ConfluenceResponse<Attachment> getPageAttachments(String pageId)
+			throws Exception {
+		return getPageAttachments(pageId, 0, 50);
+	}
+
+	/**
+	 * <p>Get the attachments of the given page using pagination</p>
+	 * @param pageId the page id
+	 * @param start The start value to get attachments from
+	 * @param limit The number of attachments to get from start
+	 * @return a {@code ConfluenceResponse} instance containing the attachment results and some pagination values</p>
+	 * @throws Exception
+	 */
+	public ConfluenceResponse<Attachment> getPageAttachments(String pageId, int start,
+			int limit) throws Exception {
+		String url = String.format("%s://%s:%s/%s/%s/%s%s?limit=%s&start=%s",
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code uses &#39;%s&#39; to format integers: limit (declared at line 350), start (declared at line 350). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 360. Consider formatting this integer with &#39;%d&#39; instead.